Quick fix for arviz import

There is a recent issue with arviz import. 

- [x] a quick fix is implemented based on https://discourse.pymc.io/t/importerror-cannot-import-name-gaussian-from-scipy-signal/14170.
- [x] changes made to print more information about arviz import error for debugging purpose

example of updated error msg (now the second line about scipy error shows):
```
E           ImportError: The arviz package is required for this functionality. Please make sure arviz is installed. See below for the original error message:
E           cannot import name 'gaussian' from 'scipy.signal' (/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/scipy/signal/__init__.py)
cuqi/samples/_samples.py:24: ImportError
```

--- a/cuqi/samples/_samples.py
+++ b/cuqi/samples/_samples.py
@@ -17,7 +17,7 @@ except ImportError as e:
 def _check_for_arviz():
     if arviz is None:
         msg = "The arviz package is required for this functionality. "\
-            + "Please make sure you have a working installation of arviz`. "
+            + "Please make sure you have a working installation of arviz`."
         raise ImportError(msg) from arviz_import_error
 
 

--- a/cuqi/samples/_samples.py
+++ b/cuqi/samples/_samples.py
@@ -17,12 +17,8 @@ except ImportError as e:
 def _check_for_arviz():
     if arviz is None:
         msg = "The arviz package is required for this functionality. "\
-            + "Please install arviz using `pip install arviz`. "\
-            + "See below for the original error message:\n"
-        arviz_import_error.args = (
-            msg + arviz_import_error.args[0],
-        ) + arviz_import_error.args[1:]
-        raise arviz_import_error
+            + "Please make sure you have a working installation of arviz`. "
+        raise ImportError(msg) from arviz_import_error
 
 
 class Samples(object):

--- a/cuqi/samples/_samples.py
+++ b/cuqi/samples/_samples.py
@@ -17,8 +17,11 @@ except ImportError as e:
 def _check_for_arviz():
     if arviz is None:
         msg = "The arviz package is required for this functionality. "\
-            + "Please make sure you have a working installation of arviz`."
-        raise ImportError(msg) from arviz_import_error
+            + "Please make sure arviz is installed. "\
+            + "See below for the original error message:\n"\
+            + arviz_import_error.args[0]
+
+        raise ImportError(msg)
 
 
 class Samples(object):

--- a/cuqi/samples/_samples.py
+++ b/cuqi/samples/_samples.py
@@ -16,11 +16,9 @@ except ImportError as e:
 
 def _check_for_arviz():
     if arviz is None:
-        msg = (
-            "The arviz package is required for this functionality. "
-            + "Please install arviz using `pip install arviz`. "
+        msg = "The arviz package is required for this functionality. "\
+            + "Please install arviz using `pip install arviz`. "\
             + "See below for the original error message:\n"
-        )
         arviz_import_error.args = (
             msg + arviz_import_error.args[0],
         ) + arviz_import_error.args[1:]

--- a/cuqi/samples/_samples.py
+++ b/cuqi/samples/_samples.py
@@ -8,13 +8,23 @@ from copy import copy
 from numbers import Number
 
 try:
-    import arviz # Plotting tool
-except ImportError:
+    import arviz  # Plotting tool
+except ImportError as e:
     arviz = None
+    arviz_import_error = e
+
 
 def _check_for_arviz():
     if arviz is None:
-        raise ImportError("The arviz package is required for this functionality. Please install arviz using `pip install arviz`.")
+        msg = (
+            "The arviz package is required for this functionality. "
+            + "Please install arviz using `pip install arviz`. "
+            + "See below for the original error message:\n"
+        )
+        arviz_import_error.args = (
+            msg + arviz_import_error.args[0],
+        ) + arviz_import_error.args[1:]
+        raise arviz_import_error
 
 
 class Samples(object):

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 matplotlib
 numpy>=1.17.0
-scipy
+scipy<1.13
 arviz

